tools/code-tools/asmtools.sh

```diff
@@ -1,80 +1,58 @@
-
-    echo "Building asmtools"
-
-    ant build
-
-    # WORKSPACE/asmtools/build/BUILD_DIR
-    echo "Moving down to ${BUILD_DIR}"
-    cd "${BUILD_DIR}"
-
-    echo "Asmtools artifact = $artifact"
-
-    echo "Copying release/lib/asmtools*.jar file to ../"
-    cp release/lib/asmtools*.jar ..
-
-    # WORKSPACE/asmtools/build/BUILD_DIR/dist
-    echo "Moving into $artifact-build/dist"
-    cd dist
-
-    echo "tar-ing ${artifact}.zip into ${artifact}.tar"
-    tar fcv "$artifact.tar" *.zip
-
-    echo "Moving ${artifact}.tar to ../.."
-    mv "$artifact.tar" ../..
-    # WORKSPACE/asmtools/build/
-    cd ../..
-
-    echo "gzipping ${artifact}.tar"
-    gzip -9 -f "${artifact}.tar"
-
-    if [ -d "$artifact-build" ];
-    then
-        mv "$artifact-build" asmtools/
-    fi
-
-    echo "Creating checksums for asmtools.jar and ${artifact}.tar.gz"
-    sha256sum asmtools.jar > asmtools.jar.sha256sum.txt
-    sha256sum "${artifact}.tar.gz" > "${artifact}.tar.gz.sha256sum.txt"
-
-    echo "Moving into asmtools"
-    cd asmtools
+    local  branchOrTag=$1
+    export JAVA_HOME=$2
+    git checkout $branchOrTag
+    echo "Moving into maven build..."
+    pushd maven
+      #export JAVA_TOOL_OPTIONS="-Dfile.encoding=UTF8 -Xdoclint:none -Dmaven.javadoc.skip=true -Dgpg.skip"
+      echo "Generating asmtools maven wrapper"
+      sh mvngen.sh
+      echo "Cleaning possible previous run"
+      mvn clean
+      echo "Running asmtools tests"
+      mvn test || echo "Test now correctly fails, this have to be fixed upstream"
+      echo "Running asmtools build"
+      mvn package -DskipTests # mvn install will do much more, but I doubt we wish that (javadoc, sources, gpg sign...)
+      echo "Moving down to target"
+      pushd target
+        echo "Asmtools artifact:"
+        ls -l
+        local mainArtifact=`ls asmtools*.jar`
+        echo "Copying maven/target/$mainArtifact file to WORKSPACE($WORKSPACE)"
+        cp  $mainArtifact $WORKSPACE
+        if [ -e surefire-reports ] ; then
+          local testResults=`echo $mainArtifact | sed "s/.jar/-tests.tar.gz/"`
+          echo "Compressing and archiving test results as $testResults"
+          tar -czf $testResults surefire-reports
+          echo "Copying maven/target/$testResults file to WORKSPACE($WORKSPACE)"
+          cp  $testResults $WORKSPACE
+        else
+          echo "No test results!"
+        fi
+      popd
+    popd
 }
 
-cd asmtools
+if [ ! -e asmtools ] ; then
+  git clone https://github.com/openjdk/asmtools.git
+fi
 
-export PRODUCT_VERSION=$(grep "PRODUCT_VERSION     \= " build/productinfo.properties | awk '{print $3}')
-# shellcheck disable=SC2005,SC2046
-export BUILD_DIR=$(echo $(eval echo $(grep "BUILD_DIR = " build/build.properties | awk '{print $3}')))
+pushd asmtools
+  latestRelease=`git tag -l | tail -n 2 | head -n 1`
+  generateArtifact "master" /usr/lib/jvm/java-1.8.0-openjdk
+  generateArtifact "at8" /usr/lib/jvm/java-17-openjdk
```

Comment:
Maybe double check the JAVA_HOME for jdk17 in adoptopenjdk/centos7_build_image. I believe it should be `/usr/lib/jvm/jdk-17.0.4.1+1`? 

Otherwise it failed with

```
The JAVA_HOME environment variable is not defined correctly
This environment variable is needed to run this program
NB: JAVA_HOME should point to a JDK not a JRE
```

Reply:
I see. So there is hardcodec jdk17? How does it changes?  Thanx for pointing up.. I'm terribly used for rhel/fedora naming conventions, and utterly forget that this is centos7. Will update. TY!

Reply:
updated